### PR TITLE
SAAS-4562: Added Infrastructure Configuration Notification Enabled Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    getInfrastructureConfiguration.data.infrastructureConfiguration.snsTopicArn) {
+                    helpers.addResult(results, 0,
+                        'Imagebuilder has infrastructure configuration notifications enabled',
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        'Imagebuilder does not have infrastructure configuration notifications enabled',
+                        region, resource);
+                }
+
```
Comment (sadeed12345) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:63]:
```suggestion
```

File: helpers/aws/regions.js [hunk 2]
```diff
@@ -193,5 +193,6 @@ module.exports = {
     appmesh:  ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'ca-central-1', 'eu-central-1',
         'eu-west-1', 'eu-west-2', 'eu-west-3', 'eu-north-1', 'ap-northeast-1', 'ap-northeast-2',
         'ap-southeast-1', 'ap-southeast-2', 'ap-south-1', 'sa-east-1', 'ap-east-1', 'me-south-1'],
-    frauddetector: ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1', 'ap-southeast-1', 'ap-southeast-2']
+    frauddetector: ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1', 'ap-southeast-1', 'ap-southeast-2'],
+    imagebuilder: regions
```
Comment (AkhtarAmir) [inline helpers/aws/regions.js:197]: Add new regions

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Infrastructure Configuration Notification Enabled',
+    category: 'Imagebuilder',
+    domain: 'compute',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:7]:
```suggestion
    domain: 'Compute',
```

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    severity: 'LOW',
+    description: 'Ensure that Imagebuilder infrastructure configuration have sns notifications enabled.',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:9]:
```suggestion
    description: 'Ensure that Image Builder infrastructure configurations have SNS notifications enabled.',
```

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    more_info: 'Infrastructure configurations allow you to specify the infrastructure within which to build and test your EC2 Image Builder image.',
+    link: 'https://docs.aws.amazon.com/imagebuilder/latest/userguide/manage-infra-config.html',
+    recommended_action: 'Enable sns notification in Imagebuilder infrastructure configurations to get notified of any changes in the service',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:12]:
```suggestion
    recommended_action: 'Enable SNS notification in EC2 Image Builder infrastructure configurations to get notified of any changes in the service.',
```

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                var getInfrastructureConfiguration = helpers.addSource(cache, source,
+                    ['imagebuilder', 'getInfrastructureConfiguration', region, recipe.arn]);
+
+                if (!getInfrastructureConfiguration || getInfrastructureConfiguration.err || !getInfrastructureConfiguration.data) {
+                    helpers.addResult(results, 3,
+                        `Unable to get infrastructure configuration description: ${helpers.addError(getInfrastructureConfiguration)}`,
+                        region, resource);
+                    continue;
+                } 
+
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:51]:
```suggestion
```

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+               
+                if (getInfrastructureConfiguration.data.infrastructureConfiguration && 
+                    getInfrastructureConfiguration.data.infrastructureConfiguration.snsTopicArn) {
+                    helpers.addResult(results, 0,
+                        'Imagebuilder has infrastructure configuration notifications enabled',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:56]:
```suggestion
                        'Infrastructure configuration has SNS notifications enabled',
```

File: plugins/aws/imagebuilder/infraConfigNotificationEnabled.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        'Imagebuilder does not have infrastructure configuration notifications enabled',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/infraConfigNotificationEnabled.js:60]:
```suggestion
                        'Infrastructure configuration does not have SNS notifications enabled',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: helpers/aws/regions_china.js [hunk 9]
```diff
@@ -117,7 +117,8 @@ module.exports = {
     voiceid: regions,
     appmesh: regions,
     frauddetector: regions,
-    imagebuilder: regions,
+    imagebuilder: regions
```
Comment (AkhtarAmir) [inline helpers/aws/regions_china.js:120]:
```suggestion
    imagebuilder: regions,
```

File: helpers/aws/regions_gov.js [hunk 10]
```diff
@@ -116,7 +116,8 @@ module.exports = {
     voiceid: regions,
     appmesh: regions,
     frauddetector: regions,
-    imagebuilder: regions,
+    imagebuilder: regions
```
Comment (AkhtarAmir) [inline helpers/aws/regions_gov.js:119]:
```suggestion
    imagebuilder: regions,
```

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**